DS/DS4/GuestHouse.java

```diff
@@ -0,0 +1,2575 @@
+			{
+				rooms.put(i, "");	// initially each room is empty
+			}
+			
+			
+			System.out.println("All Rooms are full now");
+			System.out.println("Status of rooms is: "); // at last print status of all rooms
+			roomNumbers = rooms.keys();
+			while(roomNumbers.hasMoreElements()) {
+		         roomNumber = roomNumbers.nextElement();
+		    do{
+				flag = true;
+				System.out.println("1. Show Status of rooms");	// create menu for user option
+				System.out.println("2. Allot room to a guest");
+				System.out.println("3. Particular Room Status");
+				System.out.println("4. Particular Person Status");
+				System.out.println("5. Exit");
+				System.out.println("Enter your choice: ");
+				do {
+					System.out.println("(Please Enter Positive integer Number less than 6)");	// take user choice
+				while(!scan.hasNextInt()) {
+					System.out.println("Please Enter Integer Number");
+					scan.next();
+				}
+				choice = scan.nextInt();
+				} while( choice <= 0 || choice >= 6 );
+				switch(choice)
+				{
+					case 1:
+						System.out.println("Status of rooms is: ");
+						System.out.println(roomAlotementObject.showRoomStatus(roomNumbers,rooms,roomNumber));
+					    break;
+				    
+					case 2:
+						System.out.println("Enter name of guest: ");
+						name = bufferedReaderObj.readLine(); // take the name of guest
+						System.out.println("Enter age of the guest: ");
+						do {
+							System.out.println("(Please Enter Positive integer Number greater than or equal to 18)");
+						while(!scan.hasNextInt()) {
+							System.out.println("Please Enter Integer Number");
+							scan.next();			// take age of guest
+						}
+						age = scan.nextInt();
+						} while(age < 18);
+						System.out.println(roomAlotementObject.allotRoom(rooms,age,name,numberOfRooms));
+						break;
+						
+					case 3:
+						System.out.println("Enter Room Number: ");
+						do {
+							System.out.println("(Please Enter Positive integer Number)");
+						while(!scan.hasNextInt()) {
+							System.out.println("Please Enter Integer Number");
+							scan.next();
+						}
+						roomNumber = scan.nextInt();
+						} while(roomNumber < 0 || roomNumber > (numberOfRooms - 1)); // check room status
+						System.out.println(roomAlotementObject.getPerticularRoomStatus(roomNumber, rooms));
+						break;
+						
+					case 4:
+						System.out.println("Enter Name: ");
+						roomNumber = 0;
+						name = bufferedReaderObj.readLine();// take name of guest
+						System.out.println(roomAlotementObject.getPerticularPersonStatus(name, rooms, roomNumbers, roomNumber));
+						break;
+						
+					case 5:
+						System.out.println("System Exited");
+						System.exit(0);
+						break;
+						
+					default:
+						System.out.println("Enter correct Choice");
+				}
+			} while(count < numberOfRooms); // continue the loop until all rooms is full
+		         System.out.println("Room "+(roomNumber) + " : " +
+		         rooms.get(roomNumber));
+			}
+			scan.close();
```

Comment:
Closing of resources like this should be handled only in the finally block. As finally block is always executed even if the program ended in some un-handled  exception.